### PR TITLE
Updated loss.py and metrics.py to allow serialisation (added test scr…

### PR DESCRIPTION
**Pull request by stephengmatthews**
Hello -- Serialization and deserialization were not working so this PR intends to fix that. The key things that have changed are:

1. Variables in `OrdinalCrossEntropy()` have changed to allow for serialization (specifically `num_classes` is no longer optional and `importance_weights` is a dtype that can be serialized).
2. The metric now subclasses from tf.keras to allow for serialization by implementing `get_config()`.
3. Fixed incorrect calculation of MAE for labels by using `tf.squeeze()` to ensure same rank tensors are passed to subtraction. Note: `axis=-1` was not needed in `tf.reduce_mean()` on L44 of metrics.py.
4. Added a test class to help with the above. I've also tested with my own model and the extended examples in colab linked to from the `README.md`.

What are your thoughts?

### CODE REVIEW

**Comment by ck37** [comment]
Thank you so much, really appreciate it! I’m on vacation right now but will do a quick double-check on Wed/Thu

-ck.mobile

**Comment by ck37** [comment]
@stephengmatthews Would you like to be a co-author on this package? I think this is enough contribution to warrant that.

**Comment by stephengmatthews** [comment]
> @stephengmatthews Would you like to be a co-author on this package? I think this is enough contribution to warrant that.

Thanks @ck37 -- yes, I'd like to help out!
